rust_src/src/buffers.rs

```diff
@@ -1020,10 +1018,10 @@ pub fn erase_buffer() {
     }
 }
 
-pub fn per_buffer_idx(offset: isize) -> isize {
-    let flags = LispBufferRef::new(unsafe { &mut buffer_local_flags as *mut Lisp_Buffer });
-    let flags_object = flags.as_ptr() as *const LispObject;
-    unsafe { (*(flags_object.add(offset as usize))).as_fixnum_or_error() as isize }
+pub unsafe fn per_buffer_idx(offset: isize) -> isize {
+    let flags = &mut buffer_local_flags as *mut Lisp_Buffer as *mut libc::c_char;
+    let obj = flags.add(offset as usize) as *const LispObject;
```

Comment:
Since `offset` is an isize, you should probably just use the `offset` method rather than the `add` method.

Reply:
ok.